### PR TITLE
Feat/#960 cont int setup

### PR DESCRIPTION
**Pull request by ciaranschutte**
CI jenkinsfile based on dac-ui

### CODE REVIEW

File: Jenkinsfile.groovy [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+ */
+
+def dockerRegistry = "ghcr.io"
+def githubRepo = "icgc-argo/rdpc-ui"
+def commit = "UNKNOWN"
+def version = "UNKNOWN"
+
+pipeline {
+    kubernetes {
+        label 'platform-ui-executor'
```
Comment (ciaranschutte) [inline Jenkinsfile.groovy:28]: Need a label for a kubernetes instance

File: Jenkinsfile.groovy [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+            }
+        }
+
+        stage('Deploy to argo-dev') {
+            when {
+                branch "develop"
+            }
+            steps {
+                container('docker') {
+                    withCredentials([usernamePassword(credentialsId: 'argoContainers', usernameVariable: 'USERNAME', passwordVariable: 'PASSWORD')]) {
```
Comment (ciaranschutte) [inline Jenkinsfile.groovy:97]: i think this injects credentials from our kubernetes env?

File: Jenkinsfile.groovy [hunk 3]
```diff
@@ -0,0 +1,143 @@
...
+                        sh "docker login ${dockerRegistry} -u $USERNAME -p $PASSWORD"
+                    }
+                    sh "docker tag ${dockerRegistry}/${githubRepo}:${commit} ${dockerRegistry}/${githubRepo}:${version}-${commit}"
+                    sh "docker push ${dockerRegistry}/${githubRepo}:${version}-${commit}"
+                }
+                build(job: "/ARGO/provision/platform-ui", parameters: [
```
Comment (ciaranschutte) [inline Jenkinsfile.groovy:103]: provision job passing vars

Comment (ciaranschutte) [reply Jenkinsfile.groovy:103]: need to create a provision job for rdpc-ui vs platform-ui

File: Jenkinsfile.groovy (hunk 1 above)
Comment (justincorrigible) [reply Jenkinsfile.groovy:28]: Better yet, that attribute has been deprecated. 🌮 
```suggestion
```

File: Jenkinsfile.groovy (hunk 2 above)
Comment (justincorrigible) [reply Jenkinsfile.groovy:97]: Close! they're injected from the Jenkins environment.
These for instance are simple Github credentials for an account with permissions to the Argo org.

**Comment by ciaranschutte** [comment]
some odd stuff with how the jenkinsfile is being formatted - looking into it

**Comment by ciaranschutte** [comment]
rouge semi colon in the yaml config - fixed
CI failing is ok, things are working as expected, no "test" script currently for the `npm t` command

**Pull request merged**